security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -425,7 +425,9 @@ func (sc *SecretManagerClient) generateKeyCertFromExistingFiles(certChainPath, k
 		}
 		return err
 	}
-	if err := b.RetryWithContext(context.TODO(), secretValid); err != nil {
```

Comment:
This was intended to try till secret is valid (unless the file does not exist) - Is this causing CPU issue? If file does not exist is it not coming out? I guess readWithTimeout has the problem?

Reply:
The root cause is the backoff lib that fall into dead loop

Reply:
This might send invalid combination of key and cert to Envoy and get rejected.

Reply:
We have validated via `tls.LoadX509KeyPair(certChainPath, keyPath)`

Reply:
I know but that is what `secretValid` does and it was added to ensure if key is written and cert is not yet or has some issues. Is secretValid being called repeatedly is the issue? But **exponential** backoff should have helped right?

Reply:
>Is secretValid being called repeatedly is the issue?

This is the indirect cause.  The root is ExponentialBackOff.MaxElapsedTime = 15 minutes, after that `NextBackOff` returns -1, then retry fall into dead loop.

Reply:
Is it possible to fix it such a way that when ExponentialBackOff.MaxElapsedTime - next backoff is not called and still uses MaxElapsedTime?

Reply:
Sorry, not sure what you mean. Is it not to retry after maxElapsed time?

Reply:
I mean when MaxInterval is reached, https://github.com/istio/istio/blob/20bb4cad57d6abc5d30d9adb819321d47527909c/pkg/backoff/exponential.go#L74 and NextBackOff returns -1, just use MaxInterval

Reply:
IC, that sounds good to me.